src/pathfora.js

```diff
@@ -948,26 +917,65 @@
         utils.addClass(widget, 'pf-no-img');
       }
 
-      widgetMessage.innerHTML = config.msg;
+      switch(config.type) {
+      case 'sitegate':
+      case 'form':
 
-      if (config.type === 'form') {
-        if (config.nameField === false) {
-          // FIXME Cache
-          widgetForm.removeChild(widgetForm.querySelector('input[name="username"]'));
-        }
-        if (config.titleField === false) {
-          // FIXME Cache
-          widgetForm.removeChild(widgetForm.querySelector('input[name="title"]'));
-        }
-        if (config.emailField === false) {
-          // FIXME Cache
-          widgetForm.removeChild(widgetForm.querySelector('input[name="email"]'));
+        if (config.showSocialLogin === false) {
+          node = widget.querySelector('.pf-social-login');
+
+          if (node.parentNode) {
+            node.parentNode.removeChild(node);
+          }
+        } else {
+          core.autoCompleteFacebookData();
+          core.autoCompleteGoogleData();
         }
-        if (config.msgField === false) {
-          // FIXME Cache
-          widgetForm.removeChild(widgetForm.querySelector('texarea[name="message"]'));
+
+        var getFormElement = function(field) {
+          if (field === "message")
+            return widget.querySelector('textarea');
+          else if (field === "name")
+            return widget.querySelector('input[name="username"]');
+          return widget.querySelector('input[name="' + field + '"]');
         }
+
+        // Set placeholders
+        Object.keys(config.placeholders).forEach(function (inputField) {
+          var element = getFormElement(inputField);
+
+          if (element) {
+            element.placeholder = config.placeholders[inputField];
```

Comment:
i dont think this one will cause an error but is it safe to assume these are the right type to have placeholder or do we need to validate that its a field/input etc? might be over kill as long as it doesn't break on error.